src/realm/object-store/collection.hpp

```diff
@@ -0,0 +1,104 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2020 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#ifndef REALM_OS_COLLECTION_HPP
+#define REALM_OS_COLLECTION_HPP
+
+#include <realm/collection.hpp>
+#include <realm/object-store/property.hpp>
+#include <realm/object-store/util/copyable_atomic.hpp>
+
+namespace realm {
+class Realm;
+class Results;
+class ObjectSchema;
+
+namespace object_store {
+class Collection {
+public:
+    // The Collection object has been invalidated (due to the Realm being invalidated,
+    // or the containing object being deleted)
+    // All non-noexcept functions can throw this
+    struct InvalidatedException : public std::logic_error {
+        InvalidatedException()
+            : std::logic_error("Access to invalidated List object")
+        {
+        }
+    };
+
+    // The input index parameter was out of bounds
+    struct OutOfBoundsIndexException : public std::out_of_range {
+        OutOfBoundsIndexException(size_t r, size_t c);
+        size_t requested;
+        size_t valid_count;
+    };
+
+    const std::shared_ptr<Realm>& get_realm() const
+    {
+        return m_realm;
+    }
+    // Get the type of the values contained in this List
+    PropertyType get_type() const
+    {
+        return m_type;
+    }
+
+    // Get the ObjectSchema of the values in this List
+    // Only valid if get_type() returns PropertyType::Object
+    ObjectSchema const& get_object_schema() const;
```

Comment:
Would it be possible to slowly nudge the Object Store towards the same coding style as the rest of the code base?

Reply:
Absolutely. I hate this syntax.